### DISCUSSION TITLE
I have to reinstall the GitInfo package in order for it to work

### DISCUSSION THREAD

**Discussion by david-wsd** [devlooped/GitInfo#223] [Q&A]
## Describe the Bug

GitInfo does not create ThisAssembly.Git.g.cs after package restore. I have to reinstall GitInfo for it to work.

## Steps to Reproduce

1)  Install the GitInfo Nuget package to a cs project.
As expected, ThisAssembly.Git.g.cs gets successfully created.
2)  Commit the changes to git.
3)  Clone the solution in another folder, and build.
Visual Studio automatically restores GitInfo. However for me, ThisAssembly.Git.g.cs doesn't get created. and won't get created until reinstalling GitInfo, like so:
Update-Package GitInfo -reinstall

Has anyone else encountered this issue?
Any insight as to why reinstall works, but restore doesn't?

## Additional Info

I'm developing in c#, .NET Framework 4.6.2

**Comment by kzu** [comment]
If you do a rebuild, it should still do the right thing. Could you provide more info on what .NET SDK/VS versions are you using and so on? This is working flawlessly on many many projects, so it may be some weird combination.

**Reply by david-wsd** [reply]
Hi @kzu, thanks for the response. I still really want this to work! I'm running this config:

- c# projects targeting .NET Framework 4.6.2
- Visual Studio Professional 2019 version 16.11.18
- Nuget Package manager 5.11.2

**Reply by kzu** [reply]
Is the project using SDK style?

**Reply by david-wsd** [reply]
I hadn't heard of SDK style before. Are you referring to [this](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/overview)? If so, no I don't see any indication in my project files that they're SDK style.

I think what I'll do is simply create a new project from scratch and see if that resolves my issue. I'm assuming it will, since GitInfo is so widely used. I'll let you know the outcome.

**Reply by kzu** [reply]
Migrating to SDK-style will cleanup your project SO MUCH you won't believe it. It will also reduce to virtually zero any merge conflicts at the project level across the team too. If you don't have much customizations in your .csproj, the [project migration tool](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-migrate) should Just Work.

**Reply by david-wsd** [reply]
Well that sounds pretty nice. I'll check it out, thanks!
